lib/Fmt_ast.ml

```diff
@@ -776,11 +776,11 @@ and fmt_core_type c ?(box = true) ?pro ?(pro_space = true) ?constraint_ctx
     | _ -> Fn.id
   in
   match ptyp_desc with
-  | Ptyp_alias (typ, txt) ->
+  | Ptyp_alias (typ, str) ->
       hvbox 0
         (parenze_constraint_ctx
            ( fmt_core_type c (sub_typ ~ctx typ)
-           $ fmt "@ as@ " $ fmt_type_var txt ) )
+           $ fmt "@ as@ " $ fmt_type_var str.txt ) )
```

Comment:
We need to check the comments at the location `str.loc` here

Reply:
I have pushed a commit that I believe fixes and tests this issue.